From 5de748e983c81cac6105cd2cd0aa3f6d9f394359 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Thu, 16 Jan 2025 08:58:45 +0100
Subject: [PATCH 1548/1695] product/tfm-stm32mp2: add
 CFG_SCPFW_MOD_POWER_DOMAIN flag

Put power domain modules under CFG_SCPFW_MOD_POWER_DOMAIN flags.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 61892e652232414ee74ba68df2b90f7ab9ae24dc product/tfm-stm32mp2: add CFG_SCPFW_MOD_POWER_DOMAIN flag]
Change-Id: I8fa390e514c985296f45c78d1633fe41821890f9
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/437356
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 .../SCP-firmware/product/tfm-stm32mp2/fw/Firmware.cmake         | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/Firmware.cmake b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/Firmware.cmake
index cf65d6704..6f3299264 100644
--- a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/Firmware.cmake
+++ b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/Firmware.cmake
@@ -55,7 +55,9 @@ list(APPEND SCP_MODULES "scmi-voltage-domain")
 list(APPEND SCP_MODULES "tfm-reset")
 list(APPEND SCP_MODULES "reset-domain")
 list(APPEND SCP_MODULES "scmi-reset-domain")
+if(CFG_SCPFW_MOD_POWER_DOMAIN)
 list(APPEND SCP_MODULES "power-domain")
 list(APPEND SCP_MODULES "stm32-pd")
 list(APPEND SCP_MODULES "scmi-power-domain")
+endif(CFG_SCPFW_MOD_POWER_DOMAIN)
 list(APPEND SCP_MODULES "tfm-console")
-- 
2.47.2

